src/common.ts

```diff
@@ -4,9 +4,7 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import { basename } from 'path';
-import { QueryResult } from 'jsforce';
-import { AuthInfo, AuthFields, Logger, SfdcUrl, SfdxProject, Messages, Org, SfdxError, sfdc } from '@salesforce/core';
+import { AuthInfo, Logger, SfdcUrl, SfProject, Messages, SfError } from '@salesforce/core';
```

Comment:
handleSideEffects is available in core as `AuthInfo#handleAliasAndDefaultSettings`

You may want to check core [sfdcUrl.ts](https://github.com/forcedotcom/sfdx-core/blob/v3/src/util/sfdcUrl.ts) to see if any of the resolveLoginUrl function can be refactored.